engine/apps/public_api/views/schedules.py

```diff
@@ -120,3 +126,92 @@ def export(self, request, pk):
         # Not using existing get_object method because it requires access to the organization user attribute
         export = ical_export_from_schedule(self.request.auth.schedule)
         return Response(export, status=status.HTTP_200_OK)
+
+    @action(methods=["get"], detail=True)
+    def final_shifts(self, request, pk):
+        schedule = self.get_object()
+
+        if not isinstance(schedule, OnCallScheduleWeb):
+            return Response(
+                "OnCall shifts exports are currently only available for web calendars",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        start_date_field_name = "start_date"
+        end_date_field_name = "end_date"
+
+        def _field_is_required(field_name):
+            return Response(f"{field_name} is required", status=status.HTTP_400_BAD_REQUEST)
+
+        def _field_is_invalid_date(field_name):
+            return Response(
+                f"{field_name} is not a valid date, must be in any valid ISO 8601 format",
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        def _convert_date(value):
+            if not value:
+                return None
+
+            try:
+                return date.fromisoformat(value)
+            except ValueError:
+                return None
+
+        start_date_str = request.query_params.get(start_date_field_name, None)
+        start_date = _convert_date(start_date_str)
+
+        end_date_str = request.query_params.get(end_date_field_name, None)
+        end_date = _convert_date(end_date_str)
+
+        if start_date_str is None:
+            return _field_is_required(start_date_field_name)
+        elif start_date is None:
+            return _field_is_invalid_date(start_date_field_name)
+        elif end_date_str is None:
+            return _field_is_required(end_date_field_name)
+        elif end_date is None:
+            return _field_is_invalid_date(end_date_field_name)
```

Comment:
maybe something like this?

```suggestion
        class FinalShiftQueryParamsSerializer(serializers.Serializer):
            start_date = serializers.DateField(required=True)
            end_date = serializers.DateField(required=True)

            def validate(self, attrs):
                if attrs["start_date"] > attrs["end_date"]:
                    raise serializers.ValidationError("start_date must be less than or equal to end_date")

                if attrs["end_date"] - attrs["start_date"] > timezone.timedelta(days=365):
                    raise serializers.ValidationError("The difference between start_date and end_date must be less than one year (365 days)")

                return attrs

        serializer = FinalShiftQueryParamsSerializer(data=request.query_params)
        serializer.is_valid(raise_exception=True)

        start_date = serializer.validated_data["start_date"]
        end_date = serializer.validated_data["end_date"]
```